pkg/controller/controllercmd/cmd.go

```diff
@@ -43,6 +44,10 @@ type ControllerCommandConfig struct {
 	// DisableServing disables serving metrics, debug and health checks and so on.
 	DisableServing bool
 
+	// LeaderElectionResourceLock sets the resource lock. If not set, controllercmd.ConfigMapsLeasesResourceLock will
+	// be used for backwards compatibility.
+	LeaderElectionResourceLock leaderElectionResourceLock
```

Comment:
we need to know whether it's `lease` or `configmapsleases`, so can we make it a boolean
```
// DEPRECATED: at some point everything should uses lease by default
OverrideConfigmapsLeasesResourceLock bool
```

This will allow new operators to opt in for `lease`